### ISSUE TITLE
[Bug]: styleUrl and custom font: Mapbox error MapLoad error Failed to load glyphs

### ISSUE THREAD

**Issue by remigallego** [rnmapbox/maps#3348]
### Mapbox Implementation

Mapbox

### Mapbox Version

10.16.2

### Platform

iOS, Android

### `@rnmapbox/maps` version

10.1.3

### Standalone component to reproduce

```javascript
import React from 'react';
import {
  MapView,
  ShapeSource,
  LineLayer,
  Camera,
} from '@rnmapbox/maps';

const aLine = {
  type: 'LineString',
  coordinates: [
    [-74.00597, 40.71427],
    [-74.00697, 40.71527],
  ],
};

const styleUrl = "mapbox://styles/remigallego/clrq7yzzt00au01pebcy438s3"

class BugReportExample extends React.Component {
  render() {
    return (
         <MapView
            styleUrl={styleUrl}
          >
        <Camera centerCoordinate={[-74.00597, 40.71427]} zoomLevel={14} />
        <ShapeSource id="idStreetLayer" shape={aLine}>
          <LineLayer id="idStreetLayer" />
        </ShapeSource>
      </MapView>
    );
  }
}
```

### Observed behavior and steps to reproduce

- I added a custom font to my mapbox custom style in Studio.
- I added the styleUrl to my MapView component
- The style is correctly displayed, however the fonts are not, and I get the following error message:

```
Mapbox error Map load failed: {type=glyphs, message=Failed to load glyphs: HTTP status code 404, begin=20384103955} {"level": "error", "message": "Map load failed: {type=glyphs, message=Failed to load glyphs: HTTP status code 404, begin=20384103955}", "tag": "RCTMGLMapView"}
```

**Comment by mfazekas** [comment]
@remigallego it's likely an issue with your style or an upstream issue with `mapbox-maps-ios` or `mapbox-maps-android`. Nothing the react-native wrappers can do about this.

**Comment by dziter** [comment]
@remigallego : I had same issue. Then, I realized that the font name in the Mapbox Studio is not the same than my font file (.ttf)

1 - Go to [Mapbox studio](https://studio.mapbox.com/)
2 - Create a dummy style for testing purpose
3 - Click on top right, on the navbar to 'Fonts' to find your font name
4 - On React Native side, to use the font on your marker label style:
```
const markerLabelStyle = {
  ...
  textFont: ['fontName']
}
```

**Comment by Oyzuu** [comment]
Hi, it seems like it doesn't work with custom fonts.
I'm facing the same issue, where I can't find a way to specify a custom "glyphs" URL as specified in the documentation — [Glyphs](https://docs.mapbox.com/style-spec/reference/glyphs/).

I tried with the `styleJSON`prop on the `MapView`component but nothing gives.

Is there a way to add an additional style sheet/definition to include these custom glyphs?

**Comment by paperrin** [comment]
@Oyzuu We've been able to add custom fonts by doing this in our app (tested on Android ONLY, we don't develop for iOS):

1. Manually fetch the `styles.json` file that you would have put in `MapboxGL.MapView`'s `styleURL` prop, which should look something like this (this is the messiest part I think...)
```json
 "glyphs": "mapbox://fonts/whatever/{fontstack}/{range}.pbf",
 ```
2. (optional) If you want to keep the fonts of the map intact, fetch all of the fonts from the `glyphs` property of the `style.json` at runtime
3. Programmatically modify the `glyphs` property to point to a custom path (on the device), where you'll put your font files (pbf format) that matches the variable glyph path:
`"/local/path/on/your/device/{fontstack}/{range}.pbf"`
4. Download your custom font files to that folder at runtime (we have them downloaded from our server to the same path, as you can't specify 2 different paths....), but you could find other ways...
5. Put the `style.json` object with the altered `glyphs` property in the `styleJSON` prop of the `MapboxGL.MapView`

The custom font is then used in that way (in the `style` prop of a a `SymbolLayer`:
```json
{
  textFont: [fontface]
}
```
Where `fontface` matches the name of the folder of your font in your device local files.

ALTERNATIVELY (and that's how we first started to test it...), you can modify the `glyphs` property of the `style.json` to point to `assets://{fontstack}/{range}.pbf`, and put the font files (pbf) in these paths: `android/app/src/main/assets/{fontface}/{range}.pbf`.

Works for us, even though this feels very hacky and we're waiting to find a better way to do this...

**Comment by puopg** [comment]
Wanna just add my 2 cents with loading fonts on a SymbolLayer.

I had my font uploaded to my mapbox studio map, and although all the things on the map itself like road names had the font, the symbol layer I was rendering didnt have my custom font. Tried to use the name of the font, and nothing.

Well turns out it was just that easy, but the name was just wrong. I downloaded the map configuration from studio, and looked at the JSON. In there, I saw font names for the other aspects of the map I had configured. I tried one of those names, and boom the 404 error went away and the font looked correct on the SymbolLayer.

So if you have custom fonts on other parts of the primary map, try this approach to see if its just a naming thing.
